api/ai/interfaces/algorithm_options.py

```diff
@@ -128,8 +128,14 @@ def get_schema() -> Schema:
                 SchemaOptional("enemy_behaviour"): Or(
                     *[behaviour.value for behaviour in RelationshipBehaviour]
                 ),
-                SchemaOptional("attributes_to_diversify"): Or(List[int], []),
-                SchemaOptional("attributes_to_concentrate"): Or(List[int], []),
+                SchemaOptional("attributes_to_diversify"): And(
+                    list,
+                    lambda _data: all(isinstance(i, int) for i in _data),
+                ),
+                SchemaOptional("attributes_to_concentrate"): And(
+                    list,
+                    lambda _data: all(isinstance(i, int) for i in _data),
+                ),
```

Comment:
I'm curious
1. What was the issue with the old types?
2. How is it related to production deployment?

Reply:
This was supposed to be a separate PR but I put it here to conveniently deploy it. It has been a while, but I think the reason for this was this throws an error with `List` because it does not recognize it and wants to use `list` instead.